File: BlueSTSDK/src/main/java/com/st/BlueSTSDK/Utils/advertise/BleAdvertiseInfo.java
package com.st.BlueSTSDK.Utils.advertise;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.st.BlueSTSDK.Node;

public interface BleAdvertiseInfo {
    @NonNull String getName();

    byte getTxPower();

    @Nullable String getAddress();

    int getFeatureMap();

    byte getDeviceId();

    short getProtocolVersion();

    Node.Type getBoardType();

    boolean isBoardSleeping();

    boolean isHasGeneralPurpose();
}

File: st_blue_sdk/src/main/java/com/st/blue_sdk/bt/advertise/BleAdvertiseInfo.kt
/*
 * Copyright (c) 2022(-0001) STMicroelectronics.
 * All rights reserved.
 * This software is licensed under terms that can be found in the LICENSE file in
 * the root directory of this software component.
 * If no LICENSE file comes with this software, it is provided AS-IS.
 */
package com.st.blue_sdk.bt.advertise

import com.st.blue_sdk.models.Boards

interface BleAdvertiseInfo {

    fun getName(): String

    fun getTxPower(): Byte

    fun getAddress(): String?

    fun getFeatureMap(): Long

    fun getOptionBytes(): Long

    fun getDeviceId(): Byte

    fun getProtocolVersion(): Short

    fun getBoardType(): Boards.Model

    fun isBoardSleeping(): Boolean

    fun hasGeneralPurpose(): Boolean
}